crates/sui-indexer/src/handlers/mod.rs

```diff
@@ -44,3 +55,107 @@ pub struct EpochToCommit {
     pub new_epoch: IndexedEpochInfo,
     pub network_total_transactions: u64,
 }
+
+#[async_trait]
+pub trait Handler<T>: Send + Sync {
+    async fn start_transform_and_load(
+        &self,
```

Comment:
Before we jump into creating another trait for this...we already have a trait that is a part of the ingestion framework `Worker` that should be the core trait that each of these independent data pipelines implement

Reply:
@bmwill the current `worker` trait only has `process_checkpoint` 
- which does not cover the orchestrator logic of this function, thus `start_objects_snapshot_processor` was needed for objects_snapshot, here the codes can be shared by all handlers
-  it does cover committing to DB either, thus `commit_objects_snapshot` is needed, and can be consolidated now.

Reply:
with that being said, it's great if ingestion framework can extend further for this, while ingestion framework is adopted more than just indexer and so might takes more thoughts and cycles to change.

Reply:
things like `get_watermark_hi`, `set_watermark_hi`, `get_checkpoint_lag_limiter` usually are part of ProgressStore.
Maybe we should add a primitive reducer functionality to ingestion core itself too.
That's not unique to sui indexer

Reply:
no it's not, while the main benefit of the trait here is the shared implementation of `start_transform_and_load`, so that each segregated pipeline does not have do this again.

Reply:
data ingestion framework is more than just indexer, so to achieve segregated indexer pipelines, I would prefer to focus on the indexer splitting work instead of worrying about other dependencies of ingestion framework.

Reply:
the segregation is not a huge project ,it's at most a couple of weeks', so if possible I would like to finish this first and  consolidate with phoenix with enhanced version of ingestion framework.